calblueprint/meepanyar-node
# Script to update Customer's "Starting Meter Reading"

This PR introduces a script that should be run monthly to update the starting meter of every customer at the start of each period.

To Test:
1. Comment out the `updateCustomerStartingMeters()` on line 48 of `updateCustomerStartingMeters.js` and run the 2 test functions. The 2 test functions are linked to "[Offline Customer](https://airtable.com/tblTvgiveHJFGGS8a/viw9K68RFT7Rof7ob/recEa415yyWRft0xE?blocks=hide)" (has existing meter readings) and "[Created Customer](https://airtable.com/tblTvgiveHJFGGS8a/viw9K68RFT7Rof7ob/rec20H5LwmMok670M?blocks=hide)" (does not have existing meter readings).
2. Change the values of the "Starting Meter Reading" column for these 2 customers to any value
3. Check the "Meter Reading and Invoice" table and notice the latest meter reading for Offline Customer (also linked [here](https://airtable.com/tblCwTR0ELUxM3ajs/viwBBZgPHfGG70ktc/recQuOBG1Mtqtq3sC?blocks=hide))
4. Run the script via `npm run update-customer-starting-meters` while you have an existing backend instance running
5. Ensure the `Offline Customer` has their meter reading changed to the latest meter reading seen on step 3, and that `Created Customer`'s meter reading remains unchanged.
